### DISCUSSION TITLE
lolMiner 1.32

### DISCUSSION THREAD

**Discussion by Lolliedieb** [Lolliedieb/lolMiner-releases#727] [General]
# lolMiner 1.32
(Note: Windows build next 48 hrs)

### Changes over 1.31
- **Beta Feature:** Added RTX 3000 series semi-unlock for LHR v2 cards giving up to 30% more performance then in locked state. Use --mode LHR2 to call it (and --mode LHR1 for 3060 LHR1 cards). Also added a low power LHR mode for V2 cards (--mode LHRLP). See below for more details. **Recommended drivers** for LHR2 and LHRLP: **470.63.01 or 465.31** - others could be more unstable. Most tests were done in Linux.  Use **--lhrtune** to improve either performance or stability. Read the guide for configuring here:  https://github.com/Lolliedieb/lolMiner-releases/wiki/Nvidia-Mode-Switch-&-LHR-Semi-Unlock  (2)
- Improved performance of RTX 3060 LHR v1 semi-unlock by 2-3% depending on configuration - at same low consumption!
- Added detection of the "fan glitch" for RTX 3000 LHR cards. When the glitch is detected, the GPUs will leave the special LHR modes automatically.
- Significantly improved Ergo performance on GCN Gen 1 GPUs (e.g. HD 7970, R9 280, R7 370)
- Added Ergo kernels for Pitcairn GPUs.
- New **configuration scheme** for Et(c)hash + Zil dual mining with dual stratum! See documentation here: https://github.com/Lolliedieb/lolMiner-releases/wiki/Dual-Mining-from-1.32. When using json configuration style use "DPOOLS" with same format as "POOLS" currently.
- When a pool requests a re-connection, the miner will now do so immediately instead of waiting 1 second and no longer say the connection got "lost". 
- Added support for extra nonce subscription on Ergo stratum - this will cause less reconnects on Nicehash

### Fixes 
- Fixed an issue causing "invalid" shares on Ethash when the pool makes intensive use of variable difficulty (e.g. HiveOn, Nicehash...) (1)
- Fixed an issue that might cause the epoch to update too late when doing Eth + Zil dual stratum
- Fixed an issue causing too much stale or very late shares in Ergo
- Fixed partially defect .bat example files
- Updated complete network stack to newer libraries - for more stability.
- A lot of internal re-structuring and fixes.

(1) Thanks to my Spanish mining community for letting me know and the help to track down this issue. 
(2) Feel free to discuss good tuning values in the discussion section of this release page.

<hr /><em>This discussion was created from the release <a href='https://github.com/Lolliedieb/lolMiner-releases/releases/tag/'>lolMiner 1.32</a>.</em>

**Comment by kiaas** [comment]
works great for mining Autolykos2 on an R7 370 4GB now, 35MH/s and all shares accepted, it's great to see these old things working so well again.
I commented in the issues thread I opened prior, but zombie mode for 2GB Pitcairns like my HD 7870 seems to produce only invalid results, however.

**Comment by feculay** [comment]
How do I set the 3060 lhr1 in linux?

**Reply by feculay** [reply]
now it doesn't automatically detect them at lhr1? .I have normal nvidia and lhr1 on the rig, and since the update, it does not unlock the lhr1, look at its explanation but it does not indicate information on how to make it work. Thank you very much for your prompt response

**Reply by jgonzis** [reply]
> now it doesn't automatically detect them at lhr1? .I have normal nvidia and lhr1 on the rig, and since the update, it does not unlock the lhr1, look at its explanation but it does not indicate information on how to make it work. Thank you very much for your prompt response

You should add the --mode LHR1 to use it... I suppose a detection could arrive in next realeases ...

**Reply by feculay** [reply]
Of course I added it to --mode LHR1, but it doesn't work, and I haven't seen configuration examples in this mode? Could you please give me an example on a rig with lhr1 and nvidia without blocking? Thanks

**Reply by juangahoo** [reply]
Same problem here, i added the --mode LHR1 but my card is stuck in 25 MH. (I am using Hive OS, and I put the command in the "Extra config arguments" section fo the miner)

**Reply by feculay** [reply]
its update to 1.32a in hiveos,

**Reply by feculay** [reply]
already this problem

**Reply by juangahoo** [reply]
after update v1.32a my 3060 is working well again, in fact, the performance improved from 37.50 MH to 38.55

**Reply by feculay** [reply]
i solve, hiveos update 210913 . thanks. I also have two non-lhr and two lhr1, and it reports that it detects that they are all lhr1 with the --mode lhr1 option

**Reply by feculay** [reply]
2 do not perform lhr under normal performance, and lhr1 perform with better performance. Is there any way to indicate which is the --mode lhr1 ?, so it does not affect the non-lhr ?, thanks

**Reply by feculay** [reply]
ok, I already verified and solved !, in lhr1 it is not necessary to put --mode lhr1, the recognition is automatic. Thank you

**Reply by WH3R3LAMBO** [reply]
If I have a lhr1 3060 in a mixed rig (the only 30xx series card), Should I run the dev driver to get it to work? *HiveOS 0.6-210@210913*
![r4 gpu](https://user-images.githubusercontent.com/89282607/133102245-7d1837e8-687f-4818-8c94-5a992375d012.png)

**Comment by Lolliedieb** [comment]
It was added again with 1.31 ;)

jgonzis ***@***.***> schrieb am Mo., 13. Sept. 2021, 14:47:

**Reply by zfk110** [reply]
With two GPUs 5700xt  this works fine, but with 3rd 5500xt it's crashing it's saying" device or driver not supported, but when I go back to older minner 12.0 then all three GPUs work fine. I'm on Windows 10 64bit, IDK too much about it.

**Comment by Lolliedieb** [comment]
I mean 1.32a.

Wilke Trei ***@***.***> schrieb am Mo., 13. Sept. 2021, 14:55:

**Comment by Lolliedieb** [comment]
Please update to 1.32a, this fixed the issue.

juangahoo ***@***.***> schrieb am Mo., 13. Sept. 2021, 15:05:

**Comment by Lolliedieb** [comment]
As described on the wiki link (in release notes) you can use a comma
separated list with --mode
E.g. --mode b,b,LHR1,LHR1
If first two cards are non-LHR, but last two cards are.

feculay ***@***.***> schrieb am Mo., 13. Sept. 2021, 15:45:

**Reply by feculay** [reply]
thanks, but in this way, which you indicate, it does not work with lhr1

**Comment by 12AliH** [comment]
does this unlock work for ergo I'm struggling to achieve any differences with my LHR v2 cards

**Comment by Roman7487** [comment]
3060 V1 runs at 1.32a for exactly 40 seconds, when triggered (detected mining lock) MH / s decreases by 21 and no longer accelerates back to 38 + mh / s
![PXL_20210913_151048803](https://user-images.githubusercontent.com/83368279/133123648-b27c05eb-5b2f-46b6-851e-9f654bbf8ccf.jpg)
![PXL_20210913_152225897](https://user-images.githubusercontent.com/83368279/133123659-edba153b-8116-4643-ab72-1d1e190a494c.jpg)

**Reply by jgonzis** [reply]
Which OC are you using??... Are you working with Fixed Core and with --mode LHR1?

**Reply by Roman7487** [reply]
Hive OS, the kernel did not fix, tried with the command --mode LHR1 and without it.  cards 3060 v1

**Comment by Lolliedieb** [comment]
In this case use --lhrtune -10 to start with less aggressive settings... of
that still is not stable try -15, -20 ...

Roman7487 ***@***.***> schrieb am Mo., 13. Sept. 2021, 19:18:

**Reply by Roman7487** [reply]
thanks, --lhrtune -10 doesn't work.  But --lhrtune -15 and 20 works.

**Comment by Tonitkd** [comment]
Cada dia mejor Lolminer!

usando en una 3060 V1 --mode lhr1

![image](https://user-images.githubusercontent.com/89988162/133135818-9ab2840d-cffd-4990-a2b0-c24949bc651e.png)

Mejoro de 37  a 39 mhs

Otra pregunta, lei lo del Glitch, lo hace automatico si es que lo detecta la placa? o hay que poner algun comando para que vaya probando y lo detecte?

Muchas gracias1

**Reply by feculay** [reply]
hola que oc tenes aplicado?, y si sos tan amable, cual es la extra configuracion que aplicaste con lhr1?, gracias

**Reply by Tonitkd** [reply]
![image](https://user-images.githubusercontent.com/89988162/133140287-93cb0433-861b-4e00-97b2-eec32969c48a.png)

**Reply by feculay** [reply]
gracias, y como aplicas el lhr1? en la config del lominer?

**Reply by Tonitkd** [reply]
Lo saque, porque me bajaba el OC de las otras 2, y todavia no se como designar solo a la GPU 0. Seguire probando

**Reply by feculay** [reply]
es lhr1 o 2? las que tenes

**Reply by Tonitkd** [reply]
Es lhr1 la 3060

**Reply by feculay** [reply]
si es lhr1 no debes configurar nada, es automatico, solo debes estar seguro que tengas instalado lolminer 1.32a el 210913 de hiveos actualizacion de hoy, si es lhr2 es otra forma

**Reply by Tonitkd** [reply]
Ahhh perfecto entonces, y el tema del Glitch? es automatico si logra detectarlo?

**Reply by Ricco59163** [reply]
pour avoir le Glitch , tu ne dois plus voir la vitesse de tes ventilateurs , or là , on le vois toujours ???
j ai remarqué de t'a fligtsheet est en ETH + ZIL ???
peux tu expliqué un peux mieux t'a procédure ? plz
je fais depuis 6 mois le Glitch avec une flightsheet ERGO , ca fonctionne bien 
merci

**Comment by Gh0StZ3N** [comment]
Currently still using the 1.31 ver. Win10 with Vega 64 for ETH. Launching the .bat errors on 1.32a, saying  

"C:\Users\x99' is not recognized as an internal or external command,
operable program or batch file."

Ran as Admin it says the .exe isn't in the folder and does a search.  Is double clicking on the .bat not how to launch the new ver? Do I need to comment out part of the new .bat?

Sorry if I'm not giving needed information here. I am limited in what I understand about coding and tools like this.

Edit: Using the .bat from 1.26 with my wallet and pool info worked fine to launch 1.32 and doesn't give me any errors. Just to be clear, my first launch of 1.32a was tried using the the files right out of the zip folder, before adding any wallet or pool info.

**Reply by jgonzis** [reply]
> Currently still using the 1.31 ver. Win10 with Vega 64 for ETH. Launching the .bat errors on 1.32a, saying
> 
> "C:\Users\x99' is not recognized as an internal or external command,
> operable program or batch file."
> 
> Ran as Admin it says the .exe isn't in the folder and does a search. Is double clicking on the .bat not how to launch the new ver? Do I need to comment out part of the new .bat?
> 
> Sorry if I'm not giving needed information here. I am limited in what I understand about coding and tools like this.
> 
> Edit: Using the .bat from 1.26 with my wallet and pool info worked fine to launch 1.32 and doesn't give me any errors. Just to be clear, my first launch of 1.32a was tried using the the files right out of the zip folder, before adding any wallet or pool info.

Just change in the bat file:
set MyVariable="%CD%\lolMiner.exe" 

for:
set MyVariable=%CD%\lolMiner.exe 

Take out the 2 " from that...

**Reply by Gh0StZ3N** [reply]
That worked for running the files until I put in my pool and wallet info.

Error: Your input to --pool "--user" can not be split into a address and a port. Please use format <pool address>:<pool port>.

(Note: This error message has a typo, "into a address" should be "into an address" but it's not related to my question, just a note of something I saw.)

This is my .bat user-editable part, as copied from my working .bat from 1.26 lolMiner and added to the 1.32a .bat file.

Rem #################################
Rem ## Begin of user-editable part ##
Rem #################################

set "POOL=eth-us-west.flexpool.io:4444"
set "WALLET=0x41980301A86b4f04B071CE91C7CEB2abbf1D1917.X99OC"

set "EXTRAPARAMETERS="

Rem #################################
Rem ##  End of user-editable part  ##
Rem #################################

This pool and wallet set in the .bat worked for running 1.32a exe when I used for launching the .bat from 1.26 so I don't know why putting this Pool and wallet in to the .bat for 1.32a gives this error. Do you see how I need to edit to correct this? I need to take out the quotes in the section for :W10? Thanks for the assistance.

**Reply by jgonzis** [reply]
http://lolminer.byethost16.com there you have also a configurator that will create you the correct bat file for Windows

**Comment by feculay** [comment]
Could you please give a complete example of lhr1 configuration? Thank you

**Reply by jgonzis** [reply]
It seems with 1.32a

some --mode LHR1 in case it locks try --mode LHR1 --lhrtune -5 or --mode LHR1 --lhrtune -10 

It seems a very aggresive Core Clock could lock the GPU

**Reply by feculay** [reply]
REALLY VERY GRATEFUL for your answers, but unfortunately what you tell me, it does not work, it drags the non-lhr1, and it lowers the performance of the others, the only way to make it work is without placing some --mode. at 1.32a

**Comment by tmissoni** [comment]
Hi! Thank you for this!
I'm mining ergo, with a 3060ti V2 from Gigabyte. I use --mode LHR2 and --tune up to -40. OC are -500 clock and 2500 mem with pl 130. 
Command line logs does not write any line that mention a detection of lock, but the hashrate goes every second from 65 to 120, back to 70 and so on. Any idea why?
Thank you!

**Reply by jgonzis** [reply]
Could you try  
Core 1500
Memory what you want
PL 150

--mode LHR2 and if it is stable add --lhrtune 5 if continues--lhrtune 10

**Reply by tmissoni** [reply]
thanks I will try!

**Reply by tmissoni** [reply]
Still nothing. Same behaviour. See screenshot attached
<img width="1043" alt="Screenshot 2021-09-14 at 08 50 29" src="https://user-images.githubusercontent.com/63042679/133209433-a1e8dcc3-3209-4359-9aa2-fc74bbe2f79f.png">
<img width="1109" alt="Screenshot 2021-09-14 at 08 50 53" src="https://user-images.githubusercontent.com/63042679/133209448-e8b4090e-2d63-43bb-85ed-c6ff6fd7b470.png">

**Reply by jgonzis** [reply]
> Still nothing. Same behaviour. See screenshot attached
> <img alt="Screenshot 2021-09-14 at 08 50 29" width="1043" src="https://user-images.githubusercontent.com/63042679/133209433-a1e8dcc3-3209-4359-9aa2-fc74bbe2f79f.png">
> <img alt="Screenshot 2021-09-14 at 08 50 53" width="1109" src="https://user-images.githubusercontent.com/63042679/133209448-e8b4090e-2d63-43bb-85ed-c6ff6fd7b470.png">

You are mining ERGO, that improvement is for ETH 😊

**Reply by tmissoni** [reply]
well yes, as I initially wrote :D 
any suggestion on ergo? Why the hashrate is bouncing so much?

**Reply by jgonzis** [reply]
Could you try for the 3060ti this OC:

Core 750
Memory 1750
PL 130

I hope that make it stable to 113Mhs

**Reply by tmissoni** [reply]
it works like charm! :)
Thank you!
any way to squeeze a bit more mh/s?

**Reply by jgonzis** [reply]
Try :
Core -600
Memory 1900
Power Limit 110

**Reply by tmissoni** [reply]
<img width="1025" alt="Screenshot 2021-09-14 at 11 10 46" src="https://user-images.githubusercontent.com/63042679/133230014-6efa29f2-e26b-4686-831f-956290baa1d2.png">

Nope.

**Reply by jgonzis** [reply]
Hard... then try with less PL... to see it gets or return to the sweetpost of before

**Reply by tmissoni** [reply]
is there a logic you can share how to play between clock/memory/pl?
Thanks a lot for your immense help!

**Reply by jgonzis** [reply]
The idea is to avoid LHR jump, so is play with the lower Core possible and the PL ... this is way 750 works like a charm

**Reply by tmissoni** [reply]
I get it. but isn't a jump from 750 to -600 a bit too hard? :)

**Reply by jgonzis** [reply]
The -600 is to play with powerlimit... try -650  if gets stable

-650 / 1900 / 105... no idea if works

**Reply by tmissoni** [reply]
can I change the oc on the fly or it is better I restart the miner?

**Reply by jgonzis** [reply]
On the fly is ok

**Reply by tmissoni** [reply]
how much memory clock is relevant?

**Reply by jgonzis** [reply]
To avoid unlock minimum 1750 other need more...

**Reply by tmissoni** [reply]
sweet spot found!!
<img width="546" alt="Screenshot 2021-09-14 at 12 17 25" src="https://user-images.githubusercontent.com/63042679/133240273-d472b77c-6f5e-4608-9a3d-28fa3b7cf8e2.png">
<img width="121" alt="Screenshot 2021-09-14 at 12 18 00" src="https://user-images.githubusercontent.com/63042679/133240277-c6b420c9-b03e-4aa2-812e-65df8c29753d.png">

**Comment by Tonitkd** [comment]
Chicos, una pregunta que no me queda claro, El Glitch, funciona automatico, o hay que colocar algun parametro? actualmnte tengo estos parametros, ya que estoy haciendo dual mining

![image](https://user-images.githubusercontent.com/89988162/133155371-bc106593-2350-4988-a77f-8bb3dd969969.png)
![image](https://user-images.githubusercontent.com/89988162/133155409-1bec4ceb-1fb7-4c0b-9bf2-81b0de9d722d.png)

Esta correcto ? o falta agregar algo? es una 3060 LHR v1, como dije anteriormente mejoro el desempeño de la 3060 , de 37 se fue a 39

Muchas gracias!

**Reply by jgonzis** [reply]
Si Salta el glitch lo detectará automáticamente

**Reply by Tonitkd** [reply]
Excelente! muchas gracias!

**Comment by jrvr114** [comment]
So, noob here.... LOL!  have everything setup but want to try this with the 4g card.  Was looing for the mine_eth_4g.bat and it did not seem to come with this version.  Am I missing something? Anyplace to download that file alone? Assuming I can not simply copy\rename the mine_eth.bat file as it would probably have some advanced parameters not called out in the standard file.  thoughts\suggestions???  Thanks!

**Reply by jgonzis** [reply]
4GB Ethereum is not possible any more in Windows, also in Linux wil be very low. Depending on your model you could try other Algos: Beam, ERGO,  ETC, ...

**Reply by jrvr114** [reply]
Gotcha!  thanks!

**Comment by Tonitkd** [comment]
Logre hacer una especie de Glitch, tengo mi placa asi

![image](https://user-images.githubusercontent.com/89988162/133289941-edebf669-d309-432d-aacc-f71208414f95.png)

Pero si paso el core a 500 , lo dejo correr un rato y vuelvo a -300 me da casi 60 MHS, pero a los minutos cae a 20.

No se si sirve el aporte

**Comment by Tonitkd** [comment]
Glitch realizado

![image](https://user-images.githubusercontent.com/89988162/133306956-81959207-1fdb-4265-b19e-8c517ecf0fde.png)

**Reply by Tonitkd** [reply]
![image](https://user-images.githubusercontent.com/89988162/133306993-bd452ece-f59b-4191-9968-23f509e3269c.png)

**Reply by Tradition2** [reply]
wow 56mhs thats gr8

**Reply by Ricco59163** [reply]
explique moi la procédure plz !

**Comment by Tradition2** [comment]
does anyone know if the fan glitch works with 3060 v2?

**Reply by amerkocan** [reply]
dont think so

**Comment by zfk110** [comment]
With two GPUs 5700xt this works fine, but with 3rd 5500xt it's crashing it's saying" device or driver not supported, but when I go back to older minner 12.0 then all three GPUs work fine. I'm on Windows 10 64bit, IDK too much about it.

**Comment by Tonitkd** [comment]
> explique moi la procédure plz !

Hola como estas? simplemente subi el core a 500, lo deje un rato corriendo y luego a - 300

**Reply by Ricco59163** [reply]
https://www.youtube.com/watch?v=S7qOxB4mwgw

**Reply by Tonitkd** [reply]
Excelente, asi es, a mi no me duro mucho tiempo, pero volvere a probarlo con ese Core. Muchas gracias!

![image](https://user-images.githubusercontent.com/89988162/142673452-24e75eb3-7f13-4627-b94d-47891d413953.png)

**Comment by jgonzis** [comment]
> Excelente, asi es, a mi no me duro mucho tiempo, pero volvere a probarlo con ese Core. Muchas gracias!
> 
> ![image](https://user-images.githubusercontent.com/89988162/142673452-24e75eb3-7f13-4627-b94d-47891d413953.png)

Si el detector del fan glitch. Por, cierto pégale un vistazo al último reviews de miners en [hashrate.no/tests](url) tal vez te sorprenda y cambies el 2o minero ;-)
